fleece/cli/config/config.py

```diff
@@ -0,0 +1,318 @@
+            data = _decrypt_dict(data, stage=stage, key=key, render=render)
+    elif isinstance(data, list):
+        data = _decrypt_list(data, stage=stage, key=key, render=render)
+    return data
+
+
+def _decrypt_list(data, stage, key, render):
+    return [_decrypt_item(v, stage=stage, key=key + '[]', render=render)
+            for v in data]
+
+
+def _decrypt_dict(data, stage=None, key='', render=False):
+    key_prefix = key + '.' if key else ''
+    for k, v in data.items():
+        data[k] = _decrypt_item(v, stage=stage, key=key_prefix + k,
+                                render=render)
+    return data
+
+
+def export_config(args):
+    if os.path.exists(args.config):
+        with open(args.config, 'rt') as f:
+            config = yaml.round_trip_load(f.read())
+        config['config'] = _decrypt_dict(config['config'])
+    else:
+        config = {'keys': {env['name']: 'enter-key-name-here'
+                           for env in STATE['awscreds'].environments},
+                  'config': {}}
+    if args.json:
+        print(json.dumps(config, indent=4))
+    elif config:
+        yaml.round_trip_dump(config, sys.stdout)
+
+
+def edit_config(args):
+    filename = '.fleece_edit_tmp'
+    skip_import = False
+
+    if os.path.exists(filename):
+        p = input('A previously interrupted edit session was found. Do you '
+                  'want to (C)ontinue that session or (A)bort it? ')
+        if p.lower() == 'a':
+            os.unlink(filename)
+        elif p.lower() == 'c':
+            skip_import = True
+
+    if not skip_import:
+        with open(filename, 'wt') as fd:
+            stdout = sys.stdout
+            sys.stdout = fd
+            export_config(args)
+            sys.stdout = stdout
+
+    subprocess.call(args.editor + ' ' + filename, shell=True)
+
+    with open(filename, 'rt') as fd:
+        stdin = sys.stdin
+        sys.stdin = fd
+        import_config(args)
+        sys.stdin = stdin
+
+    os.unlink(filename)
+
+
+def render_config(args):
+    with open(args.config, 'rt') as f:
+        config = yaml.safe_load(f.read())
+    config = _decrypt_item(config, stage=args.stage, key='', render=True)
+    if args.json:
+        print(json.dumps(config['config'], indent=4))
+    elif config:
+        yaml.round_trip_dump(config['config'], sys.stdout)
+
+
+def upload_config(args):
+    awscreds = STATE['awscreds'].get_awscreds(args.stage)
+    boto_args = {
+        'aws_access_key_id': awscreds['accessKeyId'],
+        'aws_secret_access_key': awscreds['secretAccessKey'],
+        'aws_session_token': awscreds['sessionToken']
+    }
+    if args.erase:
+        erase_parameters(args.path, **boto_args)
+    else:
+        with open(args.config, 'rt') as f:
+            config = yaml.safe_load(f.read())
+        config = _decrypt_item(config, stage=args.stage, key='', render='ssm')
+        write_parameters(config['config'], config['keys'][args.stage],
```

Comment:
I think the prefix scheme here is not a good default. If I'm reading this right, the root path for SSM is the stage name, right?

If that's the case, what happens if there are multiple services deployed to the same stage name under one account? The parameters updates across services could easily overwrite one another if there are name clashes. Even worse, updating params for one service could just blow away a bunch of params for any other services at the same stage (e.g., `dev`), no?

Reply:
I'm not sure I follow. The `config['keys'][args.stage]` is pointing to the KMS key (which probably isn't necessary based on my comment [here](https://github.com/racker/fleece/pull/54/files/4307c33fd82d67982649b233b9001ba612e9f19c#r150600830). `root_path` is provided via args (required) without a default value.

Are you saying that there should be default values to force name-spacing by service and stage? Right now it is relying on the user to provide accurate input, which helps keep usage flexible and allow for different preferences on organization. Requiring a particular format would potentially help prevent misuse, but will force users into a more prescriptive usage.

Reply:
My question was more about the lack of parameter namespacing, and about how the SSM API calls--as they were written--could wipe out other parameters for other services in the same account. It seems, though, that all of the SSM stuff was removed in the latest revision.